proto/extensions.go

```diff
@@ -287,9 +287,8 @@ func ExtensionDescs(m Message) ([]*ExtensionDesc, error) {
 	mr.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
 		if fd.IsExtension() {
 			xt := fd.(protoreflect.ExtensionTypeDescriptor)
-			if xd, ok := xt.Type().(*ExtensionDesc); ok {
```

Comment:
If the type is not `*ExtensionDesc`, that means it came from a different kind of extension type, such as a dynamic type that corresponds to an extension for which the generated code is not linked into the calling program.